File: backend/src/Api/Models/Matter.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Api.Models
{
    public class Matter : BaseEntity
    {
        [Required]
        [StringLength(255)]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        [StringLength(1000)]
        public string Description { get; set; } = string.Empty;
        
        [Required]
        public MatterStatus Status { get; set; } = MatterStatus.Intake;
        
        [StringLength(255)]
        public string? AssignedEmployee { get; set; }
        
        [Required]
        public Guid CustomerId { get; set; }
        
        [ForeignKey("CustomerId")]
        public Customer Customer { get; set; } = null!;
    }
}

File: frontend/src/types/matter.ts
export const MatterStatus = {
  Unknown: 0,
  Intake: 1,
  Consultation: 2,
  Engaged: 3,
  Preparation: 4,
  Active: 5,
  PendingResolution: 6,
  Resolved: 7,
  Billing: 8,
  Closed: 9
} as const;

export type MatterStatusType = typeof MatterStatus[keyof typeof MatterStatus];

export interface Matter {
  id: string;
  name: string;
  description: string;
  status: MatterStatusType;
  statusDisplayName: string;
  assignedEmployee?: string;
  customerId: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateMatterRequest {
  name: string;
  description: string;
  status: MatterStatusType;
  assignedEmployee?: string;
}

export interface UpdateMatterRequest {
  name: string;
  description: string;
  status: MatterStatusType;
  assignedEmployee?: string;
}
